lib/devise_token_auth/rails/routes.rb

```diff
@@ -16,6 +16,9 @@ def mount_devise_token_auth_for(resource, opts)
       omniauth_ctrl          = opts[:controllers][:omniauth_callbacks] || 'devise_token_auth/omniauth_callbacks'
       unlocks_ctrl           = opts[:controllers][:unlocks] || 'devise_token_auth/unlocks'
 
+      # check for resource override
+      route                  = opts[:as] || resource.pluralize.underscore.gsub('/', '_')
```

Comment:
I'd like a test for this but otherwise can you give me an example?

Reply:
I wasn't sure how to test this in the beginning so i thought to get earlier feedback before dive into the test codebase.

So, we have a rails app with default devise sign in routes and we are migrating the frontend to an angular app which does authenticate using `devise_token_auth`.

The problem is that i cannot change the current route of authentication we are using with `devise` and i could not figure out how to tell `devise_token_auth` to use a different route instead of `/users`.

Does this makes sense?

Reply:
didn't `at` worked? it's `mount_devise_token_auth_for 'User', at: 'REPLACE ME'`

Reply:
Nope, because `at` just changes the path, it still does work on top of the same resource which i guess `devise` does something and breaks the other call around.

If i just switch `as` for `at` when i try to sign in at devise default route it gives me a `devise_token_auth` authentication issue.

Reply:
This gems works on top of `devise` so I don't know what's your issue, can you post your routes? The `as` should change the path name which is a code thing, not an API thing

Reply:
Sure. This is my setup using `as`:

```ruby
# config/routes.rb
Rails.application.routes.draw do
  devise_for :users,
             controllers: { sessions: 'devise/custom_sessions' },
             skip: :registrations,
             path: ''

  devise_scope :user do
    resource :registration,
             only: %i[edit update],
             path: :profile,
             controller: 'devise/custom_registrations',
             as: :user_registration
  end
  mount_devise_token_auth_for 'User', at: 'auth', as: 'devise_token_auth'
end
```

And this is what i get from `bin/rails routes | grep devise`:

```
                         new_user_session GET      /sign_in(.:format)                           devise/custom_sessions#new
                             user_session POST     /sign_in(.:format)                           devise/custom_sessions#create
                     destroy_user_session DELETE   /sign_out(.:format)                          devise/custom_sessions#destroy
                   edit_user_registration GET      /profile/edit(.:format)                      devise/custom_registrations#edit
                        user_registration PATCH    /profile(.:format)                           devise/custom_registrations#update
                                          PUT      /profile(.:format)                           devise/custom_registrations#update
            new_devise_token_auth_session GET      /auth/sign_in(.:format)                      devise_token_auth/sessions#new
                devise_token_auth_session POST     /auth/sign_in(.:format)                      devise_token_auth/sessions#create
        destroy_devise_token_auth_session DELETE   /auth/sign_out(.:format)                     devise_token_auth/sessions#destroy
    cancel_devise_token_auth_registration GET      /auth/cancel(.:format)                       devise_token_auth/registrations#cancel
       new_devise_token_auth_registration GET      /auth/sign_up(.:format)                      devise_token_auth/registrations#new
      edit_devise_token_auth_registration GET      /auth/edit(.:format)                         devise_token_auth/registrations#edit
           devise_token_auth_registration PATCH    /auth(.:format)                              devise_token_auth/registrations#update
                                          PUT      /auth(.:format)                              devise_token_auth/registrations#update
                                          DELETE   /auth(.:format)                              devise_token_auth/registrations#destroy
                                          POST     /auth(.:format)                              devise_token_auth/registrations#create
                      auth_validate_token GET      /auth/validate_token(.:format)               devise_token_auth/token_validations#validate_token
                             auth_failure GET      /auth/failure(.:format)                      devise_token_auth/omniauth_callbacks#omniauth_failure
                                          GET      /auth/:provider/callback(.:format)           devise_token_auth/omniauth_callbacks#omniauth_success
                                          GET|POST /omniauth/:provider/callback(.:format)       devise_token_auth/omniauth_callbacks#redirect_callbacks
                         omniauth_failure GET|POST /omniauth/failure(.:format)                  devise_token_auth/omniauth_callbacks#omniauth_failure
```

And if i take the `as` out i get this from `bin/rails routes | grep devise`:

```
                         new_user_session GET      /sign_in(.:format)                       devise/custom_sessions#new
                             user_session POST     /sign_in(.:format)                       devise/custom_sessions#create
                     destroy_user_session DELETE   /sign_out(.:format)                      devise/custom_sessions#destroy
                   edit_user_registration GET      /profile/edit(.:format)                  devise/custom_registrations#edit
                        user_registration PATCH    /profile(.:format)                       devise/custom_registrations#update
                                          PUT      /profile(.:format)                       devise/custom_registrations#update
                         new_user_session GET      /auth/sign_in(.:format)                  devise_token_auth/sessions#new
                             user_session POST     /auth/sign_in(.:format)                  devise_token_auth/sessions#create
                     destroy_user_session DELETE   /auth/sign_out(.:format)                 devise_token_auth/sessions#destroy
                 cancel_user_registration GET      /auth/cancel(.:format)                   devise_token_auth/registrations#cancel
                    new_user_registration GET      /auth/sign_up(.:format)                  devise_token_auth/registrations#new
                   edit_user_registration GET      /auth/edit(.:format)                     devise_token_auth/registrations#edit
                        user_registration PATCH    /auth(.:format)                          devise_token_auth/registrations#update
                                          PUT      /auth(.:format)                          devise_token_auth/registrations#update
                                          DELETE   /auth(.:format)                          devise_token_auth/registrations#destroy
                                          POST     /auth(.:format)                          devise_token_auth/registrations#create
                      auth_validate_token GET      /auth/validate_token(.:format)           devise_token_auth/token_validations#validate_token
                             auth_failure GET      /auth/failure(.:format)                  devise_token_auth/omniauth_callbacks#omniauth_failure
                                          GET      /auth/:provider/callback(.:format)       devise_token_auth/omniauth_callbacks#omniauth_success
                                          GET|POST /omniauth/:provider/callback(.:format)   devise_token_auth/omniauth_callbacks#redirect_callbacks
                         omniauth_failure GET|POST /omniauth/failure(.:format)              devise_token_auth/omniauth_callbacks#omniauth_failure
```

As you can see is the same but the behaviour is a bit weird. When accessing `localhost:3000` i should get redirect to `http://localhost:3000/sign_in` to sign in but instead i get redirect to `http://localhost:3000/auth/sign_in` with a json message:

```
{
  success: false,
  errors: [
    "Use POST /sign_in para efetuar o login. GET não é suportado."
  ]
}
```

If i try to access `http://localhost:3000/sign_in` directly i get a 404.

The only way i got it fixed was with the `as` option above.

Does this help?

Reply:
It's a bit weird and I think we are missing something in order to spot the issue. But this change doesn't seem to harm a lot